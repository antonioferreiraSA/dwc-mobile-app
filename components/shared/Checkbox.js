import React from 'react';
import { StyleSheet, View } from 'react-native';
import Svg, { Path } from 'react-native-svg';
import Colors from '../../constants/Colors';

export default ({ checked = false }) => {
  return (
    <View
      style={[
        styles.box,
        { backgroundColor: checked ? Colors.gray : 'transparent' },
      ]}
    >
      <Svg width={24} height={24} viewBox="0 0 24 24" fill="none">
        {checked && (
          <Path
            strokeWidth={3}
            stroke={Colors.darkerGray}
            strokeMiterlimit={10}
            strokeDashoffset={40}
            d="M22.9 3.7l-15.2 16.6-6.6-7.1"
          />
        )}
      </Svg>
    </View>
  );
};

const styles = StyleSheet.create({
  box: {
    padding: 2,
    position: 'relative',
    zIndex: 1,
    borderWidth: 2,
    borderColor: Colors.gray,
    borderRadius: 4,
  },
});
