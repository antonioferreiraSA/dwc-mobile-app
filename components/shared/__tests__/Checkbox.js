import React from 'react';
import { render } from '@testing-library/react-native';
import Checkbox from '../Checkbox';

describe('<Checkbox/>', () => {
  test('render an unchecked checkbox by default', async () => {
    const { toJSON } = render(<Checkbox />);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render an unchecked checkbox', async () => {
    const { toJSON } = render(<Checkbox checked={false} />);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render a checked checkbox', async () => {
    const { toJSON } = render(<Checkbox checked={true} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
