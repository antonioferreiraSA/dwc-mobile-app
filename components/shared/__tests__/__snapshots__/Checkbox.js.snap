// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> render a checked checkbox 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#b8bab8",
        "borderRadius": 4,
        "borderWidth": 2,
        "padding": 2,
        "position": "relative",
        "zIndex": 1,
      },
      Object {
        "backgroundColor": "#b8bab8",
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={24}
    bbWidth={24}
    fill="none"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={24}
  >
    <RNSVGGroup
      fill={null}
      propList={
        Array [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M22.9 3.7l-15.2 16.6-6.6-7.1"
        propList={
          Array [
            "stroke",
            "strokeWidth",
            "strokeDashoffset",
            "strokeMiterlimit",
          ]
        }
        stroke={4280363313}
        strokeDashoffset={null}
        strokeMiterlimit={10}
        strokeWidth={3}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`<Checkbox/> render an unchecked checkbox 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#b8bab8",
        "borderRadius": 4,
        "borderWidth": 2,
        "padding": 2,
        "position": "relative",
        "zIndex": 1,
      },
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={24}
    bbWidth={24}
    fill="none"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={24}
  >
    <RNSVGGroup
      fill={null}
      propList={
        Array [
          "fill",
        ]
      }
    />
  </RNSVGSvgView>
</View>
`;

exports[`<Checkbox/> render an unchecked checkbox by default 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#b8bab8",
        "borderRadius": 4,
        "borderWidth": 2,
        "padding": 2,
        "position": "relative",
        "zIndex": 1,
      },
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={24}
    bbWidth={24}
    fill="none"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={24}
  >
    <RNSVGGroup
      fill={null}
      propList={
        Array [
          "fill",
        ]
      }
    />
  </RNSVGSvgView>
</View>
`;
